java/src/test/java/vowpalWabbit/learner/VWActionScoresLearnerTest.java

```diff
@@ -18,71 +18,33 @@
     @Rule
     public TemporaryFolder temporaryFolder = new TemporaryFolder();
 
-    private String[][] cbADFTrain = new String[][]{
-            new String[] {"| a:1 b:0.5", "0:0.1:0.75 | a:0.5 b:1 c:2"},
-            new String[] {"shared | s_1 s_2", "0:1.0:0.5 | a:1 b:1 c:1", "| a:0.5 b:2 c:1"},
-            new String[] {"| a:1 b:0.5", "0:0.1:0.75 | a:0.5 b:1 c:2"},
-            new String[] {"shared | s_1 s_2", "0:1.0:0.5 | a:1 b:1 c:1", "| a:0.5 b:2 c:1"}
-    };
-
     @Test
-    public void testCBExplore() throws IOException {
-        String[] cbTrain = new String[]{
-                "1:2:0.4 | a c",
-                "3:0.5:0.2 | b d",
-                "4:1.2:0.5 | a b c",
-                "2:1:0.3 | b c",
-                "3:1.5:0.7 | a d"
-        };
-
-        VWActionScoresLearner vw = VWLearners.create("--quiet --cb_explore 4");
-        ActionScores[] trainPreds = new ActionScores[cbTrain.length];
-        for (int i=0; i<cbTrain.length; ++i) {
-            trainPreds[i] = vw.learn(cbTrain[i]);
-        }
-        ActionScores[] expectedTrainPreds = new ActionScores[]{
-            new ActionScores(new ActionScore[]{
-                new ActionScore(0, 0.9625f),
-                new ActionScore(1, 0.0125f),
-                new ActionScore(2, 0.0125f),
-                new ActionScore(3, 0.0125f)
-            }),
-            new ActionScores(new ActionScore[]{
-                new ActionScore(0, 0.0125f),
-                new ActionScore(1, 0.9625f),
-                new ActionScore(2, 0.0125f),
-                new ActionScore(3, 0.0125f)
-            }),
-            new ActionScores(new ActionScore[]{
-                new ActionScore(0, 0.0125f),
-                new ActionScore(1, 0.9625f),
-                new ActionScore(2, 0.0125f),
-                new ActionScore(3, 0.0125f)
-            }),
-            new ActionScores(new ActionScore[]{
-                new ActionScore(0, 0.0125f),
-                new ActionScore(1, 0.9625f),
-                new ActionScore(2, 0.0125f),
-                new ActionScore(3, 0.0125f)
-            }),
-            new ActionScores(new ActionScore[]{
-                new ActionScore(0, 0.0125f),
-                new ActionScore(1, 0.9625f),
-                new ActionScore(2, 0.0125f),
-                new ActionScore(3, 0.0125f)
-            })
-        };
-        vw.close();
-        assertArrayEquals(expectedTrainPreds, trainPreds);
+    public void testCBADF() throws IOException {
+        testCBADF(false);
     }
 
     @Test
     public void testCBADFWithRank() throws IOException {
-        VWActionScoresLearner vw = VWLearners.create("--quiet --cb_adf --rank_all");
+        testCBADF(true);
+    }
+
+    private void testCBADF(boolean withRank) throws IOException {
+        String[][] cbADFTrain = new String[][]{
+            new String[]{"| a:1 b:0.5","0:0.1:0.75 | a:0.5 b:1 c:2"},
```

Comment:
I would replace the `,` with `\n,` so that you can see the multiline input better.